### PR TITLE
Use RATELIMIT_EXCEPTION_CLASS for configuration

### LINKED ISSUE #242: Add DRF Middleware-like support with block=True
Ref #121 Just had a need to add DRF support, and something interesting that we can do is take advantage of DRF's Throttled class. The reason we can do this is because if a user did something like this:

NOTE: Make sure `block=True` here

```python
class View(APIView):
    @method_decorator(ratelimit(rate="1/m", block=True))
    def get(self, request, *args, **kwargs):
        ...
```

Then DRF can use its default exception handler (I think; I actually haven't tested this yet...).

The implementation:

```python
from functools import wraps

from ratelimit import ALL
from ratelimit.core import get_usage
from rest_framework.exceptions import Throttled


__all__ = ("Ratelimited", "ratelimit")


class Ratelimited(Throttled):
    pass


def ratelimit(group=None, key=None, rate=None, method=ALL, block=True, message=None):
    def decorator(view):
        @wraps(view)
        def _wrapped(request, *args, **kw):
            old_limited = getattr(request, "limited", False)
            usage = get_usage(
                request=request,
                group=group,
                fn=view,
                key=key,
                rate=rate,
                method=method,
                increment=True
            )
            if usage is None:
                usage = {
                    "should_limit": False,
                    "time_left": -1,
                }
            limited = usage["should_limit"]
            request.limited = limited or old_limited
            if limited and block:
                raise Ratelimited(detail=message, wait=usage["time_left"])
            return view(request, *args, **kw)
        return _wrapped
    return decorator
```

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
Fixes #242 

A concern I have is performance. This seems like it is importing every time the view is called. I've implemented this like the rest of the package. However, I feel like this would be better if we instead made a "proxy" or global variable that we saved on first load of the class. The only problem I suppose would be dev reload.

### CODE REVIEW

File: django_ratelimit/decorators.py [hunk 1]
```diff
@@ -20,7 +23,9 @@ def _wrapped(request, *args, **kw):
                                          increment=True)
             request.limited = ratelimited or old_limited
             if ratelimited and block:
-                raise Ratelimited()
+                cls = getattr(
+                    settings, 'RATELIMIT_EXCEPTION_CLASS', Ratelimited)
+                raise (import_string(cls) if isinstance(cls, str) else cls)()
```
Comment (jsocol) [inline django_ratelimit/decorators.py:28]: Agree with doing this in general, as it makes testing far easier, but I also hear you on the performance concern. One option would be to move this into the top-level of the `decorator` function, which gets called far less often—though I admit I'm not positive that settings would _always_ be initialized already, depending when it's executed.

File: docs/settings.rst [hunk 2]
```diff
@@ -81,3 +81,9 @@ IPv4 mask for IP-based rate limit. Defaults to ``32`` (which is no masking)
 
 IPv6 mask for IP-based rate limit. Defaults to ``64`` (which mask the last 64 bits).
 Typical end site IPv6 assignment are from /48 to /64.
+
+``RATELIMIT_EXCEPTION_CLASS``
+-----------------------------
+
+A class (not instantiated) or a dotted path to a class that would be called
+on ratelimit with ``block=True``.
```
Comment (jsocol) [inline docs/settings.rst:89]:
```suggestion
A custom exception class, or a dotted path to a custom exception class, that will be
raised by ratelimit when a limit is exceeded and ``block=True``.
```

File: django_ratelimit/decorators.py (hunk 1 above)
Comment (Andrew-Chen-Wang) [reply django_ratelimit/decorators.py:28]: I think making a global variable makes the most sense where the initial variable is set to `None`. On dev reload, a new thread is created by the web server, so a django setting "reload" would essentially be a new web server and thus the global variable will be reset to `None`.

The only concern would be when this package needs to support async caching and ORM. But that's a problem for another time.

I've also got classes tomorrow, so probably won't be able to get to this until next week.

**Comment by jsocol** [comment]
Thanks for this @Andrew-Chen-Wang!

**Pull request merged**